src/app/shell/admin-tools/data/admins/create-admin/create-admin.component.ts

```diff
@@ -117,21 +158,31 @@ export class CreateAdminComponent extends CreateFormComponent implements OnInit,
   }
 
   onSubmit(): void {
-    const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
-      width: Constants.MODAL_SMALL,
-      data: {
-        type: (this.adminRole === AdminRole.ministryAdmin) ? ModalConfirmationType.createMinistryAdmin : undefined
-      }
-    });
+      const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
+        width: Constants.MODAL_SMALL,
+        data: {
+          type: this.editMode? ModalConfirmationType.updateMinistryAdmin : ModalConfirmationType.createMinistryAdmin
+        }
+      });
 
-    dialogRef.afterClosed().subscribe((result: boolean) => {
-      if (result) {
-        let ministryAdmin = new MinistryAdmin(
-          this.AdminFormGroup.value, 
-          this.AdminFormGroup.get('institution').value.id
-          );
-        this.store.dispatch(new CreateMinistryAdmin(ministryAdmin));
-      }
-    });   
+      dialogRef.afterClosed().subscribe((result: boolean) => {
+        if (result) {
+          if (this.editMode) {
+            console.log('id', this.AdminFormGroup.get('institution').value.id)
+            const ministryAdmin = new MinistryAdmin(
+              this.AdminFormGroup.value, 
+              this.AdminFormGroup.get('institution').value.id,
+              this.adminId
+              );
```

Comment:
I see you can do it before if/else, because you create a new instance anyway, so we can reduce duplication, regarding the last value, seems like you have a check if(adminId) in the constructor